File: climb2gether - backend/climb2gether - backend/climb2gether - backend/Domain/RockSchema.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace climb2gether___backend.Domain
{
    public class RockSchema
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        [ForeignKey("User")]
        public int UserId { get; set; }
        public User User { get; set; }
        public string RouteName { get; set; }
        public string RouteScale { get; set; }
        public string RouteDescription { get; set; }
        public string RouteLocation { get; set; }
        public DateTime CreationDate { get; set; }
        public bool IsPublic { get; set; }

    }
}

File: climb2gether - frontend/src/app/_models/RockSchema.ts
export class RockSchema {
    id?: number;
    userId?: number;
    routeName: string;
    routeScale: string;
    routeDescription: string;
    routeLocation: string;
    creationDate?: Date;
    imgURL?: string;
    isPublic: boolean;
    imgPath?: string;
}